lib/components_test.go

```diff
@@ -0,0 +1,39 @@
+package lib
```

Comment:
we can do either usually. it comes down to whether we want to do blackbox (only access to the exported functions of `lib`) vs whitebox (access to public and private functions of `lib`)  testing. in this case, i need to be in the `lib` package because i needed to create the `components` map via `newComponents`